include/nlohmann/detail/conversions/to_json.hpp

```diff
@@ -153,8 +153,11 @@ struct external_constructor<value_t::array>
     {
         j.m_type = value_t::array;
         j.m_value = value_t::array;
-        j.m_value.array->resize(arr.size());
-        std::copy(std::begin(arr), std::end(arr), j.m_value.array->begin());
+        if (arr.size() > 0)
+        {
+            j.m_value.array->resize(arr.size());
+            std::copy(std::begin(arr), std::end(arr), j.m_value.array->begin());
+        }
```

Comment:
What was the bug here?

Reply:
5d83e5e4e425606cb5b6ec33677a6229e0ff5c71 has the details.

Reply:
Ok, I can buy that the `begin` call should be guarded, but should the `resize` also be guarded? 

I'm worried that skipping the `resize(0)` could have an undesired side effect?

Reply:
@jaredgrubb

I'll move the resize out of the if, make sense.